keyhive_core/src/store/delegation.rs

```diff
@@ -20,32 +20,49 @@ pub struct DelegationStore<
     S: AsyncSigner,
     T: ContentRef = [u8; 32],
     L: MembershipListener<S, T> = NoListener,
->(pub(crate) Arc<Mutex<CaMap<Signed<Delegation<S, T, L>>>>>);
+> {
+    delegations: Arc<Mutex<CaMap<Signed<Delegation<S, T, L>>>>>,
+}
```

Comment:
Could we retain this as `pub(crate)`? There are times when we want to do things atomically while the lock is acquired, and taking lots of short locks for related separate actions can both introduce overhead and rules out atomicity.